pymeasure/instruments/thyracont/smartline_v2.py

```diff
@@ -0,0 +1,525 @@
+        set_process=compose_data,
+        check_set_errors=True,
+    )
+
+    display_orientation = Instrument.control(
+        "0DO00", "2DO01%i",
+        """Control the orientation of the display in relation to the pipe ('top', 'bottom').""",
+        values={"top": 0, "bottom": 1},
+        map_values=True,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    display_data = Instrument.control(
+        "0DD00", "2DD01%i",
+        """Control the display data source (strict SOURCES).""",
+        values=source,
+        cast=int,
+        map_values=True,
+        validator=validators.strict_discrete_set,
+        check_set_errors=True,
+    )
+
+    def setHigh(self, high=""):
+        """Set the high pressure to `high` pressure in mbar."""
+        self.ask_manually(2, "AH", high)
+
+    def setLow(self, low=""):
+        """Set the low pressure to `low` pressure in mbar."""
+        self.ask_manually(2, "AL", low)
+
+    " Sensor parameters"
+
+    def get_sensor_transition(self):
+        """
+        Get the current sensor transition between sensors.
+
+        return interpretation:
+            - direct
+                switch at 1 mbar.
+            - continuos
+                switch between 5 and 15 mbar.
+            - F[float]T[float]
+                switch between low and high value.
+            - D[float]
+                switch at value.
+        """
+        got = self.ask_manually(0, "ST")
+        # VSR/VSL: 0 direct switch at 1 mbar, 1 continuios between 5 to 15 mbar
+        if got == "0":
+            return "direct"
+        elif got == "1":
+            return "continuous"
+        else:
+            return got
+
+    def set_default_sensor_transition(self):
+        """Set the senstor transition mode to the default value, depends on the device."""
+        self.ask_manually(2, "ST", "1")
+
+    def set_continuous_sensor_transition(self, low, high):
+        """Set the sensor transition mode to "continuous" mode between `low` and `high` (floats)."""
+        self.ask_manually(2, "ST", f"F{low}T{high}")
+
+    def set_direct_sensor_transition(self, transition_point):
+        """Set the sensor transition to "direct" mode.
+
+        :param float transition_point: Switch between the sensors at that value.
+        """
+        self.ask_manually(2, "ST", f"D{transition_point}")
+
+    def set_sensor_transition(self, mode, *values):
+        """
+        Set the sensor transition to `mode` with optional `values`.
+
+        :param mode: "continuous", "direct"
+            Switch continuously from one sensor to another in the interval of
+            low and high. They can be set in values and default to 5, 15 for
+            VCR. Alternatively switch directly at a single piezo pressure.
+        :param values: The lower transition value and the higher transition value, if applicable.
+        """
+        if mode == "continuous":
+            if not values:
```

Comment:
For VSH the command could also be "2".

Reply:
I'm thinking about removing that method alltogether.

Reply:
fine for me. I would likely never use it. the methods above provide good alternatives